deployer/cilogon_auth.py

```diff
@@ -0,0 +1,193 @@
+import base64
```

Comment:
since we now have several auth-related python modules, maybe we should nest these all under `deployer/auth/`?

Reply:
Yes, I would be +1 on this too. _But_ happy for that to be done in a separate PR so as to not slow this one down, so long as we track it.

Reply:
I've added two lists in the top level comments to track what I believe should be happening as part of this PR and what should be their own separate issues after we merge this. WDYT?

Reply:
Looks good to me! For the docs, I think the minimal set would be the steps a team member would need to take to:

- Set up CILogon auth with a hub (that used this approach instead of `auth0`)
- Any common next steps in setting up auth (e.g. white-list one or more organizations, or specific users...basically turning our big "regexes" into more structured config)